internal/services/network/subnet_resource.go

```diff
@@ -603,6 +618,10 @@ func resourceSubnetRead(d *pluginsdk.ResourceData, meta interface{}) error {
 				d.Set("address_prefixes", props.AddressPrefixes)
 			}
 
+			if props.DefaultOutboundAccess != nil {
+				d.Set("default_outbound_access_enabled", props.DefaultOutboundAccess)
+			}
```

Comment:
(as in the Data Source) if this isn't returned (for existing resources, where this is defaulted to `true`) we should default this on our side:

```suggestion
			defaultOutboundAccessEnabled := true
			if props.DefaultOutboundAccess != nil {
				defaultOutboundAccessEnabled = *props.DefaultOutboundAccess
			}
			d.Set("default_outbound_access_enabled", defaultOutboundAccessEnabled)
```